[SYCL][ESIMD] Split tests on simd constructors into core and fp_extra

vasilytric opened the pull request:
There is no changes in the tests logic.
Tests on `simd` constructors were split into C++ data types and some specific data types:
- In the files with "*_core.cpp" at the end of the file will using C++ data types
- In the files with "*_fp_extra.cpp" at the end of the file will using specific data types

vasilytric committed:
[SYCL][ESIMD] Split tests into core and fp_extra

yuriykoch commented on SYCL/ESIMD/api/functional/type_coverage.hpp:
```diff
@@ -34,7 +34,7 @@ template <typename... T> struct named_type_pack {
 };
 
-enum class tested_types { all, fp, uint, sint };
+enum class tested_types { all, fp, fp_extra, uint, sint };
```
Would you mind to add comment stating the general rules of differentiating the types between different groups?

vasilytric replied:
I've done it [here](https://github.com/intel/llvm-test-suite/pull/748/commits/928337781bcde4e2f1e8619464de3f9df60f7b54#diff-3bbc1ab1445c827282cc08d73f76b58cf553d9ced2c799ba46e6148583896d22R37-R42)

yuriykoch replied:
So `all` doesn't include all other type groups, but only the C++ ones? Maybe we should rename `all` to the `core` (so we would have `core` and `extra` type groups) or to the `cpp`  (so we would have `cpp` and `non-cpp` type groups) then?

Nit: "all C++ signed/unsigned *integral* data types", because floating point type is signed

vasilytric replied:
> Maybe we should rename `all` to the `core`

I've done it [here](https://github.com/intel/llvm-test-suite/pull/748/commits/8d983fd269d836ba6c7611a1fca5eabc5f127235#diff-3bbc1ab1445c827282cc08d73f76b58cf553d9ced2c799ba46e6148583896d22R43)

> Nit: "all C++ signed/unsigned integral data types", because floating point type is signed

I've done it [here](https://github.com/intel/llvm-test-suite/pull/748/commits/8d983fd269d836ba6c7611a1fca5eabc5f127235#diff-3bbc1ab1445c827282cc08d73f76b58cf553d9ced2c799ba46e6148583896d22R41-R42)

yuriykoch reviewed (commented):
Some "_core.cpp" files are not marked as the "moved" ones by git. Therefore it's hard to tell if there were any changes in the test _logic_ here. Would you mind to state in PR description explicitly if there were any test logic changes or not?

vasilytric committed:
[SYCL][ESIMD] Update type_coverage.hpp

- Add comment to tested_types
- Remove sycl::half from all except tested_types::fp_extra

vasilytric committed:
[SYCL][ESIMD] Apply clang-format

vasilytric committed:
Merge branch 'intel' into add_fp_extra

vasilytric commented:
> Would you mind to state in PR description explicitly if there were any test logic changes or not?

@yuriykoch  Done

vasilytric committed:
[SYCL][ESIMD] Rename alias all into core and update comments

vasilytric commented:
@v-klochkov Would you mind to take a look?

vasilytric committed:
Merge branch 'intel' into add_fp_extra

vasilytric committed:
[SYCL][ESIMD] Apply clang-format

v-klochkov commented on SYCL/ESIMD/api/functional/ctors/ctor_fill_fp_extra.cpp:
```diff
@@ -0,0 +1,56 @@
+  bool passed = true;
+
+  const auto fp_types = get_tested_types<tested_types::fp_extra>();
+  const auto single_dim = values_pack<8>();
```
Wouldn't it be better to use 'all_dimensions' here?

vasilytric replied:
For what reason we need to run testing on all dimensions?
@kbobrovs Konst, do you have any ideas maybe?

kbobrovs replied:
Hm. I think what's there is enough. Slava is on vacation, so let's consider this resolved. He may have later comments, which can be addressed in a separate PR.

vasilytric replied:
@kbobrovs ok. thanks

v-klochkov commented on SYCL/ESIMD/api/functional/type_coverage.hpp:
```diff
@@ -34,22 +34,29 @@ template <typename... T> struct named_type_pack {
         {"char", "unsigned char", "signed char", "short", "unsigned short",
-         "int", "unsigned int", "long", "unsigned long", "float", "sycl::half",
-         "double", "long long", "unsigned long long"});
+         "int", "unsigned int", "long", "unsigned long", "float", "double",
```
I think 'double' is also one of 'extra_fp' types as some devices may require sycl::aspect checks before running 'double' features.

vasilytric replied:
I've done it in this commit: 17555715d874898214e5d6c3b0ffeb5e511408f1

v-klochkov reviewed (commented):
Splitting the tests look good. 
The main goals of such splitting are a) to have sycl::aspect checks and b) separate the types that may require special handling to smaller sub-tests.
This change addresses (b), but not (a). Is (a) planned for some near future?

vasilytric commented:
> This change addresses (b), but not (a). Is (a) planned for some near future?


@v-klochkov yes, we thinking about proposal how exactly we will do it and I think that a solution will be implemented in a separate PR

vasilytric committed:
[SYCL][ESIMD] Move double into fp_extra types category

vasilytric committed:
Merge branch 'intel' into add_fp_extra

vasilytric committed:
[SYCL][ESIMD] Desable test with denorm and ulp in fill_acc_fp_extra

vasilytric committed:
Merge branch 'intel' into add_fp_extra

vasilytric committed:
[SYCL][ESIMD] Apply clang-format

vasilytric commented:
@kbobrovs Would you mind to take a look?

CI on Windows failed due to `Basic/image/image_accessor_readsampler.cpp` test failure, but this test is not related to this PR.
CI on Windows failed due to `AtomicRef/atomic_memory_order_acq_rel.cpp` and `AtomicRef/atomic_memory_order_acq_rel_atomic64.cpp` tests failure, but these tests is not related to this PR.

vasilytric commented:
@kbobrovs All verifications is passed. Could you merge this PR?
